exchanges/okx/okx.go

```diff
@@ -0,0 +1,4908 @@
+	errMissingSubOrderType                           = errors.New("missing sub order type")
+	errMissingQuantity                               = errors.New("invalid quantity to buy or sell")
+	errDepositAddressNotFound                        = errors.New("deposit address with the specified currency code and chain not found")
+	errMissingAtLeast1CurrencyPair                   = errors.New("at least one currency is required to fetch order history")
+	errNoCandlestickDataFound                        = errors.New("no candlesticks data found")
+	errInvalidWebsocketEvent                         = errors.New("invalid websocket event")
+	errMissingValidChannelInformation                = errors.New("missing channel information")
+	errNilArgument                                   = errors.New("nil argument is not acceptable")
+	errNoOrderParameterPassed                        = errors.New("no order parameter was passed")
+	errEitherPriceVarianceOrPriceSpreadRequired      = errors.New("either \"price ratio\" or \"price variance\" is allowed to be passed")
+	errIncompleteCandlestickData                     = errors.New("incomplete candlestick data")
+	errMaxRFQOrdersToCancel                          = errors.New("no more than 100 RFQ cancel order parameter is allowed")
+)
+
+/************************************ MarketData Endpoints *************************************************/
+
+// PlaceOrder place an order only if you have sufficient funds.
+func (ok *Okx) PlaceOrder(ctx context.Context, arg *PlaceOrderRequestParam) (*PlaceOrderResponse, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	if arg.InstrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	if !(arg.TradeMode == TradeModeCross || arg.TradeMode == TradeModeIsolated || arg.TradeMode == TradeModeCash) {
+		return nil, errInvalidTradeModeValue
+	}
+	arg.Side = strings.ToUpper(arg.Side)
+	if !(arg.Side == order.Buy.String() || arg.Side == order.Sell.String()) {
+		return nil, errMissingOrderSide
+	}
+	arg.OrderType = strings.ToUpper(arg.OrderType)
+	if !(arg.OrderType == OkxOrderMarket ||
+		arg.OrderType == OkxOrderLimit ||
+		arg.OrderType == OkxOrderPostOnly ||
+		arg.OrderType == OkxOrderFOK ||
+		arg.OrderType == OkxOrderIOC ||
+		arg.OrderType == OkxOrderOptimalLimitIOC) {
+		return nil, errInvalidOrderType
+	}
+	if arg.QuantityToBuyOrSell <= 0 {
+		return nil, errInvalidQuantityToButOrSell
+	}
+	if arg.OrderPrice <= 0 && (arg.OrderType == OkxOrderLimit ||
+		arg.OrderType == OkxOrderPostOnly ||
+		arg.OrderType == OkxOrderFOK ||
+		arg.OrderType == OkxOrderIOC) {
+		return nil, fmt.Errorf("invalid order price for %s order types", arg.OrderType)
+	}
+	if !(arg.QuantityType == "base_ccy" ||
+		arg.QuantityType == "quote_ccy") {
+		arg.QuantityType = ""
+	}
+	type response struct {
+		Msg  string               `json:"msg"`
+		Code string               `json:"code"`
```

Comment:
Thank you for addressing the msg and code in `SendHttpRequest` function, these are no longer necessary in functions because you are not returning them anyway. Please remove all `msg` and `code` defined in response `structs` with exception to `SendHTTPRequest`